pkg/controller/servicebindingrequest/retriever.go

```diff
@@ -235,21 +235,29 @@ func (r *Retriever) readConfigMap(cr *unstructured.Unstructured, name string, it
 		// update cache after reading configmap/secret in cache
 		r.cache[fromPath].(map[string]interface{})[path].(map[string]interface{})[k] = value
 		// making sure key name has a configMap reference
-		r.store(cr, fmt.Sprintf("configMap_%s", k), []byte(value))
+		r.store(envPrefix, cr, fmt.Sprintf("configMap_%s", k), []byte(value))
 	}
 
 	r.Objects = append(r.Objects, u)
 	return nil
 }
 
 // store key and value, formatting key to look like an environment variable.
-func (r *Retriever) store(u *unstructured.Unstructured, key string, value []byte) {
+func (r *Retriever) store(envPrefix string, u *unstructured.Unstructured, key string, value []byte) {
 	key = strings.ReplaceAll(key, ":", "_")
 	key = strings.ReplaceAll(key, ".", "_")
-	if r.bindingPrefix == "" {
-		key = fmt.Sprintf("%s_%s", u.GetKind(), key)
+	if envPrefix == "" {
+		if r.bindingPrefix == "" {
+			key = fmt.Sprintf("%s_%s", u.GetKind(), key)
+		} else {
+			key = fmt.Sprintf("%s_%s_%s", r.bindingPrefix, u.GetKind(), key)
```

Comment:
So,
* if no envVarPrefix is provided, then we use the existing format of prefix-ing the `Kind`, else, we use the envVarPrefix.
* if an envVarPrefix is provided, we don't use the `Kind`, rather we use the provided envVarPrefix.